src/marketplace/ListingsView.js

```diff
@@ -11,19 +11,25 @@ import { allComponents } from '../globalComponents/typesAndClasses/TankComponent
 import Tank from '../tanks/Tank.js';
 import { getTanksById } from '../globalComponents/apiCalls/tankAPIIntegration';
 import TankDisplay from '../tanks/TankDisplay.js';
+import type { MarketplaceViewType } from '../globalComponents/typesAndClasses/MarketplaceViewType.js';
 
 type Props = {|
-	// This is the type of item we are buying
 	sellerType: SellingType,
 	onItemBought: () => void,
 |};
 
 type State = {|
 	userId: string,
-	// This allows for all the items that are for sale to be with in one array
+	chassisForSale: Array<SaleObject>,
+	weaponsForSale: Array<SaleObject>,
+	scannersForSale: Array<SaleObject>,
+	scannerAddonsForSale: Array<SaleObject>,
+	jammersForSale: Array<SaleObject>,
+	treadsForSale: Array<SaleObject>,
 	itemsForSale: Array<SaleObject>,
-	// This is used to hold all of the tanks for sale. Will be null when the sellerType is not tank
-	tanksForSale: ?Array<Tank>
+	tanksForSale: Array<SaleObject>,
+	tanks: Array<Tank>,
+	cards: Array<React.Element<'div'>>
```

Comment:
I haven't read everything yet, but this is almost always an indication that we are doing something wrong. We don't want to be storing how we are rendering something, we want to be storing the data that we are rendering, and then we can compute how to render it in the render() method